core/aws-core/src/test/java/software/amazon/awssdk/awscore/exception/AwsServiceExceptionTest.java

```diff
@@ -82,6 +82,60 @@ public void exceptionMessage_withoutExtendedRequestId() {
         assertThat(e.getMessage()).isEqualTo("errorMessage (Service: serviceName, Status Code: 500, Request ID: requestId)");
     }
 
+    @Test
+    public void exceptionMessage_withAttempts() {
+        AwsServiceException e = AwsServiceException.builder()
+                                                   .message("errorMessage")
+                                                   .numAttempts(6)
+                                                   .awsErrorDetails(AwsErrorDetails.builder()
+                                                                                   .errorMessage("errorMessage")
+                                                                                   .serviceName("serviceName")
+                                                                                   .errorCode("errorCode")
+                                                                                   .build())
+                                                   .statusCode(500)
+                                                   .requestId("requestId")
+                                                   .build();
+
+        assertThat(e.getMessage()).contains("(SDK Diagnostics: numAttempts = 6)");
```

Comment:
Can we verify the entire message so that we know there are no duplicated entries? I know it's not a good practice in general, but we are the owner of the client and we should make sure we emit the right logs :)